### PR TITLE
Allow to select availability zones when creating a non-BYOVPC cluster

### PR DESCRIPTION
**Pull request by oriAdler**
Currently, CS is choosing the availability zone arbitrarily for the user,
this PR adds the option to select the availability zone.

1. Add `availability-zones` flag.
2. Add a step for the interactive mode to choose an availability zone.

Related: [SDA-6222](https://issues.redhat.com/browse/SDA-6222)

Example 1:
image
Example 2:
image

### CODE REVIEW

File: cmd/create/cluster/cmd.go [hunk 1]
```diff
@@ -515,6 +527,14 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
+	isSubnetIDsSet := cmd.Flags().Changed("subnet-ids")
+	isAvailabilityZonesSet := cmd.Flags().Changed("availability-zones")
+	// Setting subnet IDs is choosing BYOVPC implicitly,
+	// and selecting availability zones is only allowed for non-BYOVPC clusters
+	if isSubnetIDsSet && isAvailabilityZonesSet {
```
Comment (oriAdler) [inline cmd/create/cluster/cmd.go:534]: I wasn't sure about this constraint. 
In CS we require both **subnets** and **availability zones** for BYOVPC cluster.

Here, if providing both, the alternative will be to validate that the subnets belonging to the availability zones provided.

If we won't allow providing both, we can prevent some user errors, as ROSA CLI is figuring out the availability zones for the subnets provided https://github.com/openshift/rosa/blob/master/cmd/create/cluster/cmd.go#L1277.

**Comment by oriAdler** [comment]
@pvasant @zgalor @ciaranRoche PTAL

**Comment by zgalor** [comment]
LGTM

File: cmd/create/cluster/cmd.go (hunk 1 above)
Comment (pvasant) [reply cmd/create/cluster/cmd.go:534]: is this a question ?

File: cmd/create/cluster/cmd.go [hunk 2]
```diff
@@ -515,6 +527,14 @@ func run(cmd *cobra.Command, _ []string) {
...
+		reporter.Errorf("Setting both the `subnet-ids` flag and the `availability-zones` flag is prohibited")
```
Comment (pvasant) [inline cmd/create/cluster/cmd.go:535]: Setting both the `subnet-ids` flag and the `availability-zones` flag is not supported

File: cmd/create/cluster/cmd.go (hunk 1 above)
Comment (pvasant) [inline cmd/create/cluster/cmd.go:534]: if isAvailabilityZonesSet  && isByoVPC {
"setting az is not supported for byo vpc. ROSA autodetects from subnet ids provided. 
}

Comment (oriAdler) [reply cmd/create/cluster/cmd.go]: Changed.

File: cmd/create/cluster/cmd.go [hunk 3]
```diff
@@ -1175,7 +1196,8 @@ func run(cmd *cobra.Command, _ []string) {
 	subnetIDs := args.subnetIDs
 	subnetsProvided := len(subnetIDs) > 0
 	r.Reporter.Debugf("Received the following subnetIDs: %v", args.subnetIDs)
-	if !useExistingVPC && !subnetsProvided && interactive.Enabled() {
+	// If the user has set the availability zones (allowed for non-BYOVPC clusters), don't prompt the BYOVPC message
+	if !useExistingVPC && !subnetsProvided && !isAvailabilityZonesSet && interactive.Enabled() {
```
Comment (oriAdler) [inline cmd/create/cluster/cmd.go:1200]: @zgalor @pvasant I've added ` && !isAvailabilityZonesSet` to the **if** statement.

**Pull request merged**